src/compile/data/optimizers.ts

```diff
@@ -78,3 +78,29 @@ export function removeUnusedSubtrees(node: DataFlowNode) {
   }
   return true;
 }
+
+/**
+ * Generates time unit nodes from selection filters and removes duplicate
+ * time unit nodes (as determined by the name of the output field)
+ * that may be generated due to selections projected over time units.
+ */
+export function cleanupSelectionTimeUnits(leaf: DataFlowNode) {
+  let fields = {};
+  return iterateFromLeaves((node: DataFlowNode) => {
+    if (node instanceof FilterNode) {
+      // Selection filters only add TimeUnitNodes on assembly.
```

Comment:
There used to be a reason that selections were parsed after datasets but unfortunately I don't recall it any longer. I just tried switching the order and all the examples seem to work correctly (perhaps the refactoring that occurred on the data side relaxed this constraint?). 

With that said, the call to generating a selection predicate for a filter transform occurs in the `expression` method which is only called during FilterNode assembly. Can this be moved to parse time as well? I can try it soon if you're not sure off the top of your head.